drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1575,6 +1431,9 @@ static enum ethernet_hw_caps eth_stm32_hal_get_capabilities(const struct device
 #endif
 #if defined(CONFIG_NET_DSA)
 		| ETHERNET_DSA_MASTER_PORT
+#endif
+#if defined(CONFIG_ETH_STM32_MULTICAST_FILTER)
+		| ETHERNET_FW_FILTER
```

Comment:
s/FW_FILTER/HW_FILTERING/

Reply:
A `build_all` test would be nice to have.

Reply:
There is the `tests/net/all` where suitable options could be added.

Reply:
@jukkar I tried to implement a `build_all` case for the stm32 driver, but it doesn't seem to take work as expected.

Could you take a look at the last commit to see what I'm missing, build output:

```log
warning: ETH_DRIVER (defined at drivers/ethernet/Kconfig:6) was assigned the value 'y' but got the
value 'n'. Check these unsatisfied dependencies: (!NET_TEST) (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_ETH_DRIVER and/or look up ETH_DRIVER in the
menuconfig/guiconfig interface. The Application Development Primer, Setting Configuration Values,
and Kconfig - Tips and Best Practices sections of the manual might be helpful too.


warning: ETH_STM32_MULTICAST_FILTER (defined at drivers/ethernet/Kconfig.stm32_hal:204) was assigned
the value 'y' but got the value 'n'. Check these unsatisfied dependencies: ETH_STM32_HAL (=n),
ETH_DRIVER (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_ETH_STM32_MULTICAST_FILTER and/or look up
ETH_STM32_MULTICAST_FILTER in the menuconfig/guiconfig interface. The Application Development
Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual
might be helpful too.


warning: PHY_INIT_PRIORITY (defined at drivers/ethernet/phy/Kconfig:16) was assigned the value '50'
but got the value ''. Check these unsatisfied dependencies: ETH_DRIVER (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_PHY_INIT_PRIORITY and/or look up
PHY_INIT_PRIORITY in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.

```